src/config.ts

```diff
@@ -1,90 +1,108 @@
-import { NetworkConfig } from './typings'
+import { NetworkConfig, RpcConfig } from './typings'
 
-let network: 'dev' | 'testnet' | 'custom' = 'dev'
+type Network = 'dev' | 'testnet'
 
-let customConfig: NetworkConfig
-
-const DEFAULT_SENTRY_DSN =
-  'https://4e27545e4faf43318301625d79a6dc34@o1011868.ingest.sentry.io/6451353'
-
-const DEV_CONFIG: NetworkConfig = {
-  AUTH_URL: 'https://verify.dev.arcana.network',
-  RPC_URL: 'https://blockchain-dev.arcana.network',
-  CHAIN_ID: '0x9dd4',
-  NET_VERSION: '40404',
-  GATEWAY_URL: 'https://gateway-dev.arcana.network',
-  WALLET_URL: 'https://wallet.dev.arcana.network',
-  SENTRY_DSN:
+const DEV_NETWORK_CONFIG: NetworkConfig = {
+  authUrl: 'https://verify.dev.arcana.network',
+  gatewayUrl: 'https://gateway-dev.arcana.network',
+  walletUrl: 'https://wallet.dev.arcana.network',
+  sentryDsn:
     'https://68615fda056a4337bcc9b7e3062562c3@o1011868.ingest.sentry.io/6449849',
 }
 
-const TESTNET_CONFIG: NetworkConfig = {
-  AUTH_URL: 'https://verify.beta.arcana.network',
-  RPC_URL: 'https://blockchain001-testnet.arcana.network/',
-  CHAIN_ID: '0x9dd5',
-  NET_VERSION: '40405',
-  GATEWAY_URL: 'https://gateway001-testnet.arcana.network',
-  WALLET_URL: 'https://wallet.beta.arcana.network',
-  SENTRY_DSN:
+const TESTNET_NETWORK_CONFIG: NetworkConfig = {
+  authUrl: 'https://verify.beta.arcana.network',
+  gatewayUrl: 'https://gateway001-testnet.arcana.network',
+  walletUrl: 'https://wallet.beta.arcana.network',
+  sentryDsn:
     'https://4e27545e4faf43318301625d79a6dc34@o1011868.ingest.sentry.io/6451353',
 }
 
-const getConfig = () => {
-  switch (network) {
-    case 'dev':
-      return DEV_CONFIG
-    case 'testnet':
-      return TESTNET_CONFIG
-    case 'custom':
-      return customConfig
-  }
+const DEV_RPC_CONFIG: RpcConfig = {
+  rpcUrls: ['https://blockchain-dev.arcana.network'],
+  chainId: 40404,
+  chainName: 'Arcana Dev',
 }
 
-function setCustomConfig(n: NetworkConfig) {
-  network = 'custom'
-  customConfig = n
-  customConfig.SENTRY_DSN = DEFAULT_SENTRY_DSN
+const TESTNET_RPC_CONFIG: RpcConfig = {
+  rpcUrls: ['https://blockchain001-testnet.arcana.network/'],
+  chainId: 40405,
+  chainName: 'Arcana testnet',
 }
 
-const setIframeDevUrl = (url: string) => {
-  DEV_CONFIG.WALLET_URL = url
+const getNetworkConfig = (n: Network | NetworkConfig) => {
+  if (typeof n === 'string' && n == 'testnet') {
+    return TESTNET_NETWORK_CONFIG
+  }
+
+  if (typeof n === 'string' && n == 'dev') {
+    return DEV_NETWORK_CONFIG
+  }
+
+  if (isNetworkConfig(n)) {
+    return n
+  } else {
+    throw new Error('Invalid network config passed')
+  }
 }
 
-const setNetwork = (n: 'testnet' | 'dev') => {
-  network = n
+const getRpcConfig = (c: RpcConfig | undefined, n: Network | NetworkConfig) => {
```

Comment:
Can simplify this a bit by using a single param. If the caller is using Network, it would be awkward to set RPC config as undefined in the first param.

```
const getRpcConfig = (c: RpcConfig | Network) => {
  if(isRpcConfig(c)) {
    ...
  }
  
  switch(c) {
    case 'testnet': ...
    case 'dev': ...
  }
}
```

Reply:
It has to be handled somewhere so it's getting handled here.
The suggestion `c: RpcConfig | Network`, how will you decide what are you passing? You will check if rpc config is not `undefined` in `InitParams` otherwise you will pass network in `c` - its the same thing happening here it's just being handled inside this function rather than some other function.